swagger.yaml

```diff
@@ -2820,36 +2882,78 @@ paths:
 definitions:
   operator:
     type: object
+    required:
+      - name
+      - operator_type
     properties:
+      name:
+        type: string
+        description: Unique name of the operator.
       operator_type:
-        $ref: '#/definitions/operator-type'
-      enabled:
-        type: boolean
-        default: false
+        type: string
+        enum: ['built-in', 'custom']
+        description: Kind of operator. Different types are monitored by the service differently.
+      timeout:
+        type: integer
+        description: Positive number represents a timeout in seconds for the operator to be available.
+
+  operator-status:
+    type: object
+    required:
+      - id
+      - cluster_id
+    properties:
+      id:
+        type: string
+        format: uuid
+        description: Unique identifier of the object.
+        x-go-custom-tag: gorm:"primary_key"
+      cluster_id:
+        type: string
+        format: uuid
+        description: The cluster that this operator is associated with.
+        x-go-custom-tag: gorm:"primary_key;foreignkey:Cluster"
       properties:
+        $ref: '#/definitions/operator'
+      report:
+        $ref: '#/definitions/operator-report'
+      status_updated_at:
         type: string
-        description: JSON-formatted string containing the properties for each operator
+        format: date-time
+        x-go-custom-tag: gorm:"type:timestamp with time zone"
+        description: Time at which the operator was last updated.
 
-  cluster-operator:
+  operator-report:
     type: object
+    description: Monitor report details of an operator.
+    required:
+      - status
     properties:
-      operator_type:
-        $ref: '#/definitions/operator-type'
-      enabled:
-        type: boolean
-        default: false
+      namespace:
+        type: string
+        description: The k8s namespace of the operator. Not all operators have a namespace.
       status:
         type: string
       status_info:
         type: string
-      properties:
-        type: string
-        description: JSON-formatted string containing the properties for each operator
+        description: Detailed information about the operator state.
 
-  supported-operators-list:
+  operators-list:
     type: array
     items:
-      $ref: '#/definitions/operator-type'
+      $ref: '#/definitions/operator'
+
+  operator-monitor-report:
```

Comment:
Why not flatten it to:

```
{
  "operator_name": "string",
  "namespace": "string",
  "status": "string",
  "status_info": "string"
}
```

Reply:
So `operator-status` would have the same latest `operator-report` information. In case it would change that we don't have to maintain fields in two places, but have a single object.

Reply:
so maybe it should be `allof`?